probe-rs/src/flashing/loader.rs

```diff
@@ -228,6 +228,26 @@ impl FlashLoader {
         Ok(())
     }
 
+    /// Prepares the data sections that have to be loaded into flash from an UF2 file.
+    /// This will validate the UF2 file and transform all its data into sections but no flash loader commands yet.
+    pub fn load_uf2_data<T: Read>(&mut self, file: &mut T) -> Result<(), FileDownloadError> {
+        let mut uf2_buffer = Vec::new();
+        file.read_to_end(&mut uf2_buffer)?;
+
+        let (converted, family_to_target) = uf2_decode::convert_from_uf2(&uf2_buffer).unwrap();
+        self.add_data(
+            family_to_target
```

Comment:
I guess we do not really have a possibility to specify a family ID. Kind of a pitty.

Reply:
So, I am kind of in favor of ignoring the family name. People using probe-rs should have a minimum understanding of embedded development to realize that binaries are not simply portable.

But it's also a bit ugly to just disregard the family ID when the standard says bootloaders should reject incompatible binaries.

Maybe let's do it simple for now and ignore it und then revisit later in a second PR if the need arises where we fix it.

I am highly against encoding dynamic values (chip names) in the probe-rs source. Target YAMLs would be the place where this goes indeed. But then again, it's called family ID but goes on a single target definition. I think MS did not think this one through when naming :D

Reply:
Perfect, thanks!